### ISSUE TITLE
Yolo 8.0.170 and higer slower then 8.0.169

### ISSUE THREAD

**Issue by Alarmod** [ultralytics/ultralytics#5221]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

I checked prediction inputs with 1) list of numpy; 2) Torch.Tensors
Yolo 8.0.169          -- 65 FPS
Yolo 8.0.170-192   -- 50 FPS

is_list modification do it

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by RizwanMunawar** [comment]
@Alarmod, could you please furnish additional details and a reproducible example regarding the problem you're facing? Providing these would greatly assist us in gaining a deeper understanding of the issue. 

Thank you.

**Comment by Alarmod** [comment]
def letterbox(im, new_shape=(640, 640), color=(114, 114, 114), auto=True, scaleFill=False, scaleup=True, stride=32):
    # Resize and pad image while meeting stride-multiple constraints
    shape = im.shape[:2]  # current shape [height, width]
    if isinstance(new_shape, int):
        new_shape = (new_shape, new_shape)

    # Scale ratio (new / old)
    r = min(new_shape[0] / shape[0], new_shape[1] / shape[1])
    if not scaleup:  # only scale down, do not scale up (for better val mAP)
        r = min(r, 1.0)

    # Compute padding
    ratio = r, r  # width, height ratios
    new_unpad = int(round(shape[1] * r)), int(round(shape[0] * r))
    dw, dh = new_shape[1] - new_unpad[0], new_shape[0] - new_unpad[1]  # wh padding
    if auto:  # minimum rectangle
        dw, dh = np.mod(dw, stride), np.mod(dh, stride)  # wh padding
    elif scaleFill:  # stretch
        dw, dh = 0.0, 0.0
        new_unpad = (new_shape[1], new_shape[0])
        ratio = new_shape[1] / shape[1], new_shape[0] / shape[0]  # width, height ratios

    dw /= 2  # divide padding into 2 sides
    dh /= 2

    if shape[::-1] != new_unpad:  # resize
        im = cv2.resize(im, new_unpad, interpolation=cv2.INTER_LINEAR)
    top, bottom = int(round(dh - 0.1)), int(round(dh + 0.1))
    left, right = int(round(dw - 0.1)), int(round(dw + 0.1))
    im = cv2.copyMakeBorder(im, top, bottom, left, right, cv2.BORDER_CONSTANT, value=color)  # add border
    return im, top, bottom, left, right, ratio, (dw, dh)

def use_buff_func(frame_buff_inside, shape):
    frame_tensor_buff = [None] * len(frame_buff_inside)
    letterboxed_frame_buff = [None] * len(frame_buff_inside)

    frame_index = 0

    top = 0
    bottom = 0
    left = 0
    right = 0
    for frame in frame_buff_inside: 
        letterboxed_frame, top, bottom, left, right, _, _ = letterbox(im=cv2.cvtColor(frame, cv2.COLOR_BGR2RGB), new_shape=shape, stride=32, auto=True) # BCHW Tensor
        letterboxed_frame_buff[frame_index] = letterboxed_frame

        frame_tensor = ToTensor()(letterboxed_frame) # Torch DEFAULT and Yolo Tensor
        frame_tensor_buff[frame_index] = torch.reshape(frame_tensor, (1, frame_tensor.shape[0], frame_tensor.shape[1], frame_tensor.shape[2])) # Torch DEFAULT and Yolo Tensor

        frame_index = frame_index + 1

    frame_tensor_buff = torch.cat(frame_tensor_buff, 0).to(device)

    # Feed frame to model and get detections
    det_start_time = time.time()
    with torch.no_grad():
         detections = model(frame_tensor_buff, agnostic_nms=True, iou=0.333, verbose=False)
    det_end_time = time.time()

**Comment by Alarmod** [comment]
Tested with python 3.9 and torch==1.13 and latest PyTorch 2, GPU Nvidia 1080 Ti

Input width and height: 720x590
ANN input width and height: 736x608 (after my letterbox correction)

use_buff_func param shape = (736x608)

**Comment by Alarmod** [comment]
I removed this code from ultralytics\models\yolo\detect\predict.py: 
`if not isinstance(orig_imgs, list):  # input images are a torch.Tensor, not a list
   orig_imgs = ops.convert_torch2numpy_batch(orig_imgs)`

My speed now 65 FPS with Torch.Tensor, Again!

**Comment by Alarmod** [comment]
def convert_torch2numpy_batch(batch: torch.Tensor) -> np.ndarray:
    return (batch.permute(0, 2, 3, 1).contiguous() * 255).clamp(0, 255).to(torch.uint8).cpu().numpy()

Data already on GPU and this function load data again to CPU

**Comment by glenn-jocher** [comment]
Thank you for your report and for the code snippet, @Alarmod. You've raised a valuable point. The `convert_torch2numpy_batch` function is indeed moving data from GPU to CPU and this could be causing the slow down you're observing.

This function was probably used to ensure compatibility across various functions and methods, which expect np.ndarray format. However, if your inputs are already on the GPU in Torch.Tensor format, converting them back to CPU format does introduce unnecessary overhead.

The team continuously strives to improve and optimize the performance of YOLOv8. We're thankful for your feedback regarding this performance bottleneck and will discuss internally possible ways to rectify it. Thank you for taking the time to provide this valuable insight, and helping improve the Ultralytics project and the broader YOLO community.
